packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx

```diff
@@ -174,48 +183,40 @@ export class TextField extends BaseComponent<ITextFieldProps, ITextFieldState> i
     const errorMessage = this._errorMessage;
 
+    this._classNames = getClassNames(styles!, {
+      theme: theme!,
+      className,
+      disabled,
+      focused: isFocused,
+      required,
+      multiline,
+      hasLabel: !!label,
+      hasErrorMessage: !!errorMessage,
+      borderless,
+      resizable,
+      hasIcon: !!iconProps,
+      underlined
+    });
+
     // If a custom description render function is supplied then treat description as always available.
     // Otherwise defer to the presence of description or error message text.
     this._isDescriptionAvailable = Boolean(this.props.onRenderDescription || description || errorMessage);
 
-    const textFieldClassName = css('ms-TextField', styles.root, className, {
-      ['is-required ' + styles.rootIsRequiredLabel]: this.props.label && required,
-      ['is-required ' + styles.rootIsRequiredPlaceholderOnly]: !this.props.label && required,
-      ['is-disabled ' + styles.rootIsDisabled]: disabled,
-      ['is-active ' + styles.rootIsActive]: isFocused,
-      ['ms-TextField--multiline ' + styles.rootIsMultiline]: multiline,
-      ['ms-TextField--underlined ' + styles.rootIsUnderlined]: underlined,
-      ['ms-TextField--borderless ' + styles.rootIsBorderless]: borderless
-    });
-
     return (
-      <div className={textFieldClassName}>
-        <div className={css('ms-TextField-wrapper', styles.wrapper, underlined ? errorMessage && styles.invalid : '')}>
+      <div className={this._classNames.root}>
+        <div className={this._classNames.wrapper}>
           {onRenderLabel(this.props, this._onRenderLabel)}
-          <div
-            className={css(
-              'ms-TextField-fieldGroup',
-              styles.fieldGroup,
-              isFocused && styles.fieldGroupIsFocused,
-              errorMessage && styles.invalid
-            )}
-          >
+          <div className={this._classNames.fieldGroup}>
             {(addonString !== undefined || this.props.onRenderAddon) && (
-              <div className={css('ms-TextField-prefix', styles.fieldPrefixSuffix)}>
-                {onRenderAddon(this.props, this._onRenderAddon)}
-              </div>
+              <div className={this._classNames.prefix}>{onRenderAddon(this.props, this._onRenderAddon)}</div>
             )}
             {(prefix !== undefined || this.props.onRenderPrefix) && (
-              <div className={css('ms-TextField-prefix', styles.fieldPrefixSuffix)}>
-                {onRenderPrefix(this.props, this._onRenderPrefix)}
-              </div>
+              <div className={this._classNames.prefix}>{onRenderPrefix(this.props, this._onRenderPrefix)}</div>
             )}
             {multiline ? this._renderTextArea() : this._renderInput()}
-            {(iconClass || iconProps) && <Icon className={css(iconClass, styles.icon)} {...iconProps} />}
+            {(iconClass || iconProps) && <Icon className={css(iconClass, this._classNames.icon)} {...iconProps} />}
```

Comment:
>iconClass [](start = 62, length = 9)

Should iconClass here be part of the baseStyles? (this was the approach I took in Dropdown)

Reply:
This is just a total miss on my part. I never intended for css function to still be used here. Thanks

Reply:
Since the `Icon` styles function is already exposed through `iconProps` I don't think there needs to be another entry for it in `subComponentStyles`.